### PR TITLE
Show minimum LBC deposit amounts for Publish / Claims

### PR DESCRIPTION
**Pull request by btzr-io**
### Fixes
Show the minimum LBC deposit amounts for Publish / Claims
> The App should either not allow something under 0.000097 or show what the minimum is.

Fix for https://github.com/lbryio/lbry-app/issues/432, https://github.com/lbryio/lbry-app/issues/440, https://github.com/lbryio/lbry-app/issues/441

### Todo
- [ ]  Get fees ( dynamically ).
- [x] Better error message.
- [x] Display minimum deposit for claim.
- [x] Display minimum bid for channel creation.
- [x] Prevent negative value from inputs.

### CODE REVIEW

**Comment by btzr-io** [comment]
minimum

**Comment by btzr-io** [comment]
min-claim

File: ui/js/component/publishForm/internal/channelSection.jsx [hunk 1]
```diff
@@ -58,6 +58,11 @@ class ChannelSection extends React.PureComponent {
         __("LBRY channel names must be at least 4 characters in length.")
       );
       return;
+    } else if (this.state.newChannelBid < this.minChannelBid()) {
+      this.refs.newChannelName.showError(
+        __(`LBRY channel bid must be at least ${this.minChannelBid()} credits.`)
```
Comment (kauffj) [inline ui/js/component/publishForm/internal/channelSection.jsx]: "The minimum amount is ${this.minChannelBid()} credits."

File: ui/js/component/publishForm/view.jsx [hunk 2]
```diff
@@ -179,6 +179,11 @@ class PublishForm extends React.PureComponent {
     return !!myClaims.find(claim => claim.name === name);
   }
 
+  minClaimBid() {
+    //Get or calculate minimum deposit to prevent: cannot be abandoned
```
Comment (kauffj) [inline ui/js/component/publishForm/view.jsx]: It's actually a bug that these cannot be abandoned :/

File: ui/js/component/publishForm/view.jsx [hunk 3]
```diff
@@ -301,8 +306,17 @@ class PublishForm extends React.PureComponent {
   }
 
   handleBidChange(event) {
+    const value = event.target.value;
+
+    if (value < this.minClaimBid()) {
+      this.refs.bid.showError(
+        __(`The minimum deposit claim is ${this.minClaimBid()} credits.`)
```
Comment (kauffj) [inline ui/js/component/publishForm/view.jsx]: Use the same error text as above.

File: ui/js/component/publishForm/internal/channelSection.jsx (hunk 1 above)
Comment (btzr-io) [inline ui/js/component/publishForm/internal/channelSection.jsx]: Ok, thanks, I'm not good with error messages :upside_down_face:

**Comment by btzr-io** [comment]
@kauffj Are negatives value allowed for some reason?

**Comment by btzr-io** [comment]
@kauffj any thought on this ?

**Comment by kauffj** [comment]
I merged this for the unused code it removed, but I did not keep the minimum checks. There is no intention for there to be minimums, and the minimums will be dropped when the corresponding bug in lbry is fixed.

I'd be open to ways we can better communicate that there is some small amount of a transaction that is unrecoverable.

**Pull request closed without merging**